Packs/MapPattern/Scripts/MapPattern/MapPattern.py

```diff
@@ -0,0 +1,194 @@
+                'output': d.get('output'),
+                'algorithm': d.get('algorithm'),
+                'next': d.get('next')
+            }
+    else:
+        raise ValueError('Invalid mappings')
+
+
+def translate(source: Any,
+              mappings: Union[List[Dict[str, Any]], Dict[str, Any]],
+              caseless: bool,
+              priority: str,
+              algorithm: str,
+              context: Any) -> Tuple[Optional[List[str]], Any]:
+    """ Replace the string given with the patterns.
+
+    :param source: The string to be replaced.
+    :param mappings: The mapping table to translate.
+    :param caseless: Set to True for caseless comparation, False otherwise.
+    :param priority: The priority order (first_match, last_match or longest_pattern).
+    :param algorithm: The default algorithm for pattern match.
+    :param context: The demisto context.
+    :return: The mapping matched and the new value replaced by it.
+    """
+    matched_mapping = None
+    matched_output = source
+    source = str(source)
+    for mapping in iterate_mapping(mappings):
+        flags = re.IGNORECASE if caseless else 0
+        algorithm = mapping.get('algorithm') or algorithm
+
+        # Check if the source matches a pattern
+        pattern = make_regex(mapping['pattern'], algorithm)
+        match = re.fullmatch(pattern, source, flags=flags)
+        if not match:
+            continue
+
+        # Check if the source matches any of exclusion patterns.
+        exclude = [make_regex(x, algorithm) for x in mapping['exclude']]
+        if any([re.fullmatch(x, source, flags=flags) for x in exclude]):
+            continue
+
+        # Set the output
+        output = mapping.get('output')
+        if output is None:
+            output = demisto.args()['value']
+        elif algorithm == 'regex' and isinstance(output, str):
+            output = match.expand(output.replace(r'\0', r'\g<0>'))
+        if isinstance(context, dict) and isinstance(output, str):
+            output = extract_value(output, extract_dt, context)
+
+        next_mappings = mapping.get('next')
+        if next_mappings:
+            mapping, output = translate(output, next_mappings, caseless, priority, algorithm, context)
+            if not mapping:
+                continue
+
+        if priority in ('first_match', 'last_match'):
+            matched_output = output
+            matched_mapping = mapping
+            if priority == 'first_match':
+                break
+        else:
+            raise ValueError(f'Invalid priority: {priority}')
+
+    return matched_mapping, matched_output
+
+
+def main():
+    args = demisto.args()
+    value = args['value']
+    mappings = args['mappings']
+    algorithm = args.get('algorithm') or DEFAULT_ALGORITHM
+    priority = args.get('priority') or DEFAULT_PRIORITY
+    caseless = argToBoolean(args.get('caseless') or 'true')
+    context = args.get('context')
+
+    if not isinstance(value, (dict, list)):
```

Comment:
so `dict` and `list` are not supported (as is in they won't be transformed)? if so, i think this should be mentioned in the argument description